framework/mutex/OracleMutex.php

```diff
@@ -0,0 +1,131 @@
+<?php
+/**
+ * @link http://www.yiiframework.com/
+ * @copyright Copyright (c) 2008 Yii Software LLC
+ * @license http://www.yiiframework.com/license/
+ */
+
+namespace yii\mutex;
+
+use PDO;
+use Yii;
+use yii\base\InvalidConfigException;
+
+/**
+ * OracleMutex implements mutex "lock" mechanism via Oracle locks.
+ *
+ * Application configuration example:
+ *
+ * ```
+ * [
+ *     'components' => [
+ *         'db' => [
+ *             'class' => 'yii\db\Connection',
+ *             'dsn' => 'oci:dbname=LOCAL_XE',
+ *              ...
+ *         ]
+ *         'mutex' => [
+ *             'class' => 'yii\mutex\OracleMutex',
+ *              ...
+ *         ],
+ *     ],
+ * ]
+ * ```
+ *
+ * @see Mutex
+ *
+ * @author Alexander Zlakomanov <zlakomanoff@gmail.com>
+ * @since 1.0
+ */
+class OracleMutex extends DbMutex
+{
+    /**
+     * @var string driver name.
+     */
+    public $driverName = 'oci';
```

Comment:
Option make absolutely no sense here: you just make develper to setup extra field in case he uses not `oci` PDO driver.

Either the check should be dropped completely or make more flexible.
For example:

```php
if (strpos($this->db->driverName, 'oci') !== 0 || strpos($this->db->driverName, 'odbc') !== 0) {
    // exception
}
```